lib/inc/internal/sources/dmi_source.hpp

```diff
@@ -52,6 +57,11 @@ namespace whereami { namespace sources {
         dmi_base() {}
         virtual ~dmi_base() {}
         /**
+         * Retrieve the BIOS address
+         * @return The BIOS address
+         */
+        std::string bios_address() const;
```

Comment:
The BIOS address is always available from dmidecode (as long as the dmidecode output isn't totally empty), but as far as I've seen, VMware is the only place it's actually useful.

The DMI source object is getting pretty crowded as more single-case values like this are collected - I feel like there's probably a better way to organize this.